android/src/main/java/com/zoontek/rnpermissions/RNPermissionsModule.java

```diff
@@ -141,16 +143,13 @@ private boolean permissionExists(final String permission) {
 
   @ReactMethod
   public void checkNotifications(final Promise promise) {
-    final boolean enabled = NotificationManagerCompat
-      .from(getReactApplicationContext()).areNotificationsEnabled();
-
-    final WritableMap output = Arguments.createMap();
-    final WritableMap settings = Arguments.createMap();
-
-    output.putString("status", enabled ? GRANTED : BLOCKED);
-    output.putMap("settings", settings);
+    if(Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU) {
+      this.checkNotificationsCompat(promise);
+      return;
+    }
 
-    promise.resolve(output);
+    String fieldName = this.getFieldName("android.permission.POST_NOTIFICATIONS");
+    this.checkPermission(fieldName, promise);
```

Comment:
The return type of `checkPermission` is `"granted" | "denied" | …`, the return type of `checkNotifications` is supposed to be `{ status: "granted" | "denied" | …, settings: {} }`

These break the TypeScript method contract